### ISSUE TITLE
Document pachd load balancing

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#2684]
There are two options here:

1) Make a publicly accessible load balancer

This will expose a public IP / DNS hostname (depending on your k8s cluster settings) to access the pachd service.

You can follow AWS instructions: https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/

2) Make an internal LB

This will allow applications deployed within the same VPC/subnet as the pach cluster to access the pachd service.

To do this:

do `kubectl edit deploy/pachd` and modify it to look like this:

```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    app: pachd
    suite: pachyderm
  name: pachyderm-service
  annotations:
      service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
spec:
  ports:
  - port: 30650
    protocol: TCP
    targetPort: 30650
  selector:
    app: pachd
    suite: pachyderm
  type: LoadBalancer
status:
  loadBalancer: {}
```
Basically make the type `LoadBalancer` and add the annotation.

If we find a link to AWS or k8s docs that explains this, we should link there from our deployment docs. However, I could only find a GH issue: https://github.com/kubernetes/kubernetes/issues/17620

**Comment by ryansmith23** [comment]
I was able to get this working for the helm chart on a private GKE cluster with an internal load balancer using a pre-defined static IP with:

```
---
kind: Service
apiVersion: v1
metadata:
  name: pachd
  labels:
    app: pachd
    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
    release: "{{ .Release.Name }}"
    heritage: "{{ .Release.Service }}"
    suite: pachyderm
  annotations:
    cloud.google.com/app-protocols: '{"api-grpc-port":"HTTP2"}'
    cloud.google.com/load-balancer-type: "Internal"
spec:
  type: LoadBalancer
  loadBalancerIP: <static-ip-address>
  ports:
    - name: api-grpc-port
      port: 650
      targetPort: 650
      nodePort: 30650
    - name: trace-port
      port: 651
      targetPort: 651
      nodePort: 30651
    - name: api-http-port
      port: 652
      targetPort: 652
      nodePort: 30652
  selector:
    app: pachd
```

**Comment by JoeyZwicker** [comment]
Thanks for the info @ryansmith23.

**Comment by mindthevirt** [comment]
Closing in favor of #3466
